### PR TITLE
feat(workflow): add graduation closeout reconciliation

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary
- add `graduation-closeout.sh` to reconcile delivered integration-branch sub-items and parent epics after graduation PR merges
- cover native sub-issue discovery, label fallback, closing-keyword refs, optional skips, partial failures, reruns, and configurable terminal status with mocked shell tests
- update Protocol 05b, command wrappers, GitHub Projects docs, smoke runbook, and changelog

## Tests
- `bash scripts/development-workflow/tests/test-graduation-closeout.sh`
- `bash scripts/development-workflow/tests/test-workflow-lib-github-projects.sh`
- `bash scripts/development-workflow/tests/test-prepare-release-tracker-cleanup.sh`
- `shellcheck scripts/development-workflow/graduation-closeout.sh scripts/development-workflow/tests/test-graduation-closeout.sh`
- `python3 scripts/lint/workflow-shell-guard-lint.py --base-ref origin/develop`
- `npx markdownlint-cli2 "docs/workflow/development-workflow/protocols/05b-graduate-development-protocol.md" "docs/workflow/development-workflow/integrations/github-projects.md" "docs/workflow/development-workflow/README.md" "docs/testing/workflow/1178-auto-close-integration-branch-subitems-on-graduation.smoke-test.md" "CHANGELOG.md" ".agents/skills/graduate-development/SKILL.md" ".claude/commands/graduate-development.md" ".cursor/commands/graduate-development.md"`
- `find docs/specs/developments docs/testing/workflow -name "*.md" -print0 | xargs -0 python3 scripts/lint/markdown-heuristic-lint.py CHANGELOG.md`
- `bash scripts/lint/check-changelog-duplicate-headers.sh CHANGELOG.md`

Closes #1178

---

> [!NOTE]
> **Medium Risk**
> The script closes issues and updates GitHub Projects via `gh`; mistakes could affect tracker state, but merged-PR validation, fail-closed discovery, explicit exclusions, and mocked tests reduce blast radius.
> 
> **Overview**
> Adds **`graduation-closeout.sh`** so post-merge graduation no longer depends on default-branch auto-close alone: after a merged `develop-<slug>` → `develop` PR, the helper discovers planned work (native sub-issues, `integration-branch:<slug>` label, closing keywords on merged sub-item PRs), closes delivered sub-items, reasserts terminal GitHub Projects status, then closes the epic unless deferred or closeout failed.
> 
> **Protocol 05b** Step 5 and **`/graduate-development`** surfaces (Claude, Cursor, Codex skill) now require running this helper and surfacing `skipped_optional` / `failed` output instead of manually closing the epic first. **BR-8/BR-9** are updated accordingly; **github-projects.md**, workflow README, smoke runbook, and **CHANGELOG** document flags (`--exclude-issue`, `--defer-epic-close`) and env-based terminal status. **`test-graduation-closeout.sh`** mocks `gh` for validation, discovery, parser edge cases, partial failure, and reruns.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b25c6fd9223c5e364dcd34247feddbfa1c439a5e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by lhpaul** [comment]
### Internal Review Gate (Codex)

Result: clean after direct fix.

Reviewed against `REVIEW.md` code-review checklist for spec compliance, shell-script validation, tracker mutation safety, parser-risk coverage, and docs consistency.

Direct fix applied during review:
- `graduation-closeout.sh` now validates that `--graduation-pr` is already merged from `develop-<slug>` to `develop` before mutating issues or tracker state.
- Added mocked regression coverage for the unmerged-PR stop path; closeout test now reports `43 passed, 0 failed`.

Verification rerun after the fix:
- `bash scripts/development-workflow/tests/test-graduation-closeout.sh`
- `shellcheck scripts/development-workflow/graduation-closeout.sh scripts/development-workflow/tests/test-graduation-closeout.sh`
- `npx markdownlint-cli2 "docs/workflow/development-workflow/protocols/05b-graduate-development-protocol.md" "docs/workflow/development-workflow/integrations/github-projects.md" "docs/testing/workflow/1178-auto-close-integration-branch-subitems-on-graduation.smoke-test.md"`
- `python3 scripts/lint/workflow-shell-guard-lint.py --base-ref origin/develop`

**Comment by lhpaul** [comment]
bugbot run

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean — no blocking findings
**Platforms:** pr-agent (clean), bugbot (clean)
**Findings:** 0 blocking, 0 suggestions

**Ready reviewer phase:** No net-new blocker was found after the draft GitHub gate.
**Ready-phase platforms:** bugbot

**Advisory findings (non-blocking):**
- Possible Issue ([view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1235#issuecomment-4981763167))
- Possible Issue ([view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1235#issuecomment-4981763167))

*Posted automatically by `pr-review-loop.sh`.*

File: scripts/development-workflow/graduation-closeout.sh [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+CLOSED_COUNT="$(count_file_lines "$CLOSED_FILE")"
+ALREADY_TERMINAL_COUNT="$(count_file_lines "$ALREADY_FILE")"
+SKIPPED_OPTIONAL_COUNT="$(count_file_lines "$SKIPPED_FILE")"
+FAILED_COUNT="$(count_file_lines "$FAILED_FILE")"
+
+EPIC_OUTPUT="$TMP_DIR/epic.out"
+EPIC_STATUS=0
+process_epic "$FAILED_COUNT" > "$EPIC_OUTPUT" || EPIC_STATUS=$?
+
+echo "GRADUATION_CLOSEOUT_RESULT=$([ "$FAILED_COUNT" -eq 0 ] && [ "$EPIC_STATUS" -eq 0 ] && echo pass || echo failed)"
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/graduation-closeout.sh:542]: Fixed in 046d0b9. `graduation-closeout.sh` now records discovery failures in the failed summary, fails closeout when no delivered sub-items can be identified, and holds the parent epic open when merged-PR discovery fails. Added mocked regressions for empty discovery and PR discovery failure; `test-graduation-closeout.sh` now reports 53 passed, 0 failed.

**Comment by lhpaul** [comment]
bugbot run

File: scripts/development-workflow/graduation-closeout.sh [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+  if ! discover_label_subitems; then
+    echo "Warning: label fallback discovery failed for ${INTEGRATION_LABEL}." >&2
+    record_result "$FAILED_FILE" "discovery" "Candidate discovery" "label:${INTEGRATION_LABEL}" "label_fallback_failed"
+  fi
+else
+  # Still include label-discovered legacy items when native sub-issues exist.
+  if ! discover_label_subitems && [ "$NATIVE_DISCOVERY_OK" -eq 0 ]; then
+    echo "Warning: label fallback discovery failed for ${INTEGRATION_LABEL}." >&2
+    record_result "$FAILED_FILE" "discovery" "Candidate discovery" "label:${INTEGRATION_LABEL}" "label_fallback_failed"
+  fi
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/graduation-closeout.sh:513]: Fixed in b25c6fd. Label fallback discovery failure is now recorded as a failed closeout condition even when native sub-issue discovery has candidates, which holds the parent epic open. Added a mocked regression for native candidates plus label discovery failure; `test-graduation-closeout.sh` now reports 58 passed, 0 failed.

**Comment by lhpaul** [comment]
bugbot run

**Comment by lhpaul** [comment]
## /run-epic PR Disposition

- Scope source: explicit item #1178 from sequential run-work batch
- Item: #1178 - workflow-process: integration-branch epic sub-items not auto-closed on graduation PR merge
- PR: #1235
- Reviewed head SHA: `b25c6fd9223c5e364dcd34247feddbfa1c439a5e`
- Reviewer result: clean
- Blocking findings: 0
- Advisory findings: 2
- Risk: medium
- Merge authority: User authorized sequential handling with delegated review and merges; effective policy --delegate-review --may-merge --may-start-backlog false --max-risk high --base develop.
- Final decision: merge_approved

### Risk Reasons

- workflow script or harness change: scripts/development-workflow/graduation-closeout.sh

### Invocation Policy

- Original command: `$run-item 1178 --delegate-review --may-merge --may-start-backlog false --max-risk high`
- Copy-paste equivalent: `$run-item 1178 --delegate-review --may-merge --may-start-backlog false --max-risk high --base develop`
- Confirmation: user requested sequential handling enabling merges until all are merged in develop

| Field | Recommended | Selected | Effective |
| --- | --- | --- | --- |
| mayStartBacklog |  |  |  |
| delegateReview | true | true |  |
| mayMerge | true | true |  |
| maxRisk | medium | high |  |
| base | develop | develop |  |

### Checkpoint Policy

Not recorded.

### Advisory Decisions

| Source | Category | Decision | Rationale |
| --- | --- | --- | --- |
| pr-agent | Possible Issue #1 | accepted | Reviewer loop classified this PR-Agent possible issue as advisory after clean review; no blocking change was required. |
| pr-agent | Possible Issue #2 | accepted | Reviewer loop classified this PR-Agent possible issue as advisory after clean review; Bugbot blocking threads were fixed and resolved; GraphQL unresolved thread count is 0. |

### Verification Evidence

- Labels: 
- CI result: unknown
- Reviewer summary present: false
- Unresolved thread count: unknown
- PR merge state: unknown
- Issue state: unknown
- Project status: unknown

### Protocol Deviations

None.
